### PR TITLE
set sealed segment to unreadable before sync target version

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
issue: #26249
 
 make sealed segment unreadable before sync target version

### CODE REVIEW

File: internal/querynodev2/delegator/distribution.go [hunk 1]
```diff
@@ -186,6 +189,9 @@ func (d *distribution) AddDistributions(entries ...SegmentEntry) {
 		if ok {
 			// remain the target version for already loaded segment to void skipping this segment when executing search
 			entry.TargetVersion = oldEntry.TargetVersion
+		} else {
+			// waiting for sync target version, to become readable
+			entry.TargetVersion = unreadableTargetVersion
```
Comment (congqixia) [inline internal/querynodev2/delegator/distribution.go:194]: what about  call from `SyncDistribution`?

Comment (weiliu1031) [reply internal/querynodev2/delegator/distribution.go:194]: sync distribution will call sync target version directly

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**